### PR TITLE
feat: projen init

### PR DESCRIPTION
**Pull request by andrestone**
*Issue #, if available:*
- https://github.com/eladb/projen/issues/10
- https://github.com/eladb/projen/issues/11

*Description of changes:*
- `projen init` initial support

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/cli.ts [hunk 1]
```diff
@@ -19,6 +15,39 @@ if (!fs.existsSync(symlink)) {
...
 }
 
-// now we are ready to rock
+const args = yargs
+  .usage('Usage: [npx / yarn] projen [<command>] [options]')
+  .example('npx projen', 'generates project files using an existing projen file')
+  .command('init <template-name>', 'initializes a new projenrc file from a template')
+  .example('npx projen init jsii', 'initializes from the jsii template')
+  // eslint-disable-next-line @typescript-eslint/no-require-imports
+  .version(require('../package.json').version)
```
Comment (eladb) [inline src/cli.ts:24]: I believe this is the default

File: src/cli.ts [hunk 2]
```diff
@@ -19,6 +15,39 @@ if (!fs.existsSync(symlink)) {
   fs.symlinkSync(projen, symlink);
 }
 
-// now we are ready to rock
+const args = yargs
+  .usage('Usage: [npx / yarn] projen [<command>] [options]')
+  .example('npx projen', 'generates project files using an existing projen file')
+  .command('init <template-name>', 'initializes a new projenrc file from a template')
```
Comment (eladb) [inline src/cli.ts:21]: Mind using `commandsDir` like here: https://github.com/awslabs/cdk8s/blob/master/packages/cdk8s-cli/bin/cdk8s.ts

File: src/cmds/init.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    // fails if .projenrc.js already exists
+    if (fs.readdirSync('.').filter(f => f !== 'node_modules').length > 0) {
+      process.stderr.write('Cannot initialize a projen project in a non-empty directory.\n');
+      process.exit(1);
+    }
+    const templatePath = path.join(templatesPath, args['TEMPLATE']);
+    try {
+      fs.copySync(templatePath, '.', {recursive: true});
+      process.stdout.write(`Projen project generated from the "${args['TEMPLATE']}" template. \n` +
+        'Now, go edit it, nothing there is what you want.\n');
```
Comment (eladb) [inline src/cmds/init.ts:25]:
```suggestion
      process.stdout.write(`Projen project generated from the "${args['TEMPLATE']}" template. \n` +
        'Now, go edit it (nothing there is what you want).\n' + 
        'When you are done, simply run "npx projen" to get the magic going.\n');
```

File: src/eslint.ts [hunk 4]
```diff
@@ -90,7 +90,8 @@ export class Eslint extends Construct {
         ignorePatterns: [
           '*.js',
           '*.d.ts',
-          'node_modules/',
+          'node_modules/', // docs say ignored it's by default?
+          'templates/**',
```
Comment (eladb) [inline src/eslint.ts:94]: This is very specific to this project, so I wouldn't put it here. We should probably add an option to add ignore patterns to `Eslint`

File: src/jsii-project.ts [hunk 5]
```diff
@@ -283,6 +283,7 @@ export class JsiiProject extends NodeProject {
         ],
         exclude: [
           'node_modules',
+          'templates',
```
Comment (eladb) [inline src/jsii-project.ts:286]: Same comment.

File: templates/typescript/.projenrc.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+const { TypeScriptLibraryProject } = require('projen');
+
+const project = new TypeScriptLibraryProject({
+    name: 'my-project',
```
Comment (eladb) [inline templates/typescript/.projenrc.js:4]: 2 indent please

File: src/eslint.ts (hunk 4 above)
Comment (andrestone) [reply src/eslint.ts:94]: Yep! Waiting for [this](https://github.com/eladb/projen/pull/4).

**Pull request merged**